### PR TITLE
MINIFICPP-2552 schema generation: simplify usage, drive-by changes

### PR DESCRIPTION
**Pull request by szaszm**
- add line ending after the last JSON line
- exit through return from main instead of `std::exit`, so destructors are called normally
- removed checks for parent directory and regular file: writing to special files or nonexistent files is totally fine, and creating the parent directory is not the responsibility of minifi.
- error handling of std::ofstream
- `--schema` now has a default argument of `-`, meaning output to stdout
- no longer chdir to MINIFI_HOME on startup (I'm not sure about this one, it may cause issues elsewhere)

----

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: minifi_main/MiNiFiMain.cpp [hunk 1]
```diff
@@ -196,6 +196,7 @@ int main(int argc, char **argv) {
     .help("Generate documentation in the specified directory");
   argument_parser.add_argument("-s", "--schema")
     .metavar("PATH")
+    .default_value("-")
```
Comment (lordgamez) [inline minifi_main/MiNiFiMain.cpp:199]: Maybe it would be better to change this argument to allow 0 or 1 values and if no value is provided then we write the schema to stdout.

**Review by adamdebreceni — APPROVED** [review]

File: minifi_main/MiNiFiMain.cpp (hunk 1 above)
Comment (szaszm) [reply minifi_main/MiNiFiMain.cpp:199]: That was my first idea too. I don't know if it's better or not, but I couldn't figure out how to do it, that's why I went with the default value approach instead. The `-` value for filename arguments is commonly used to mean stdout anyway.

Comment (lordgamez) [reply minifi_main/MiNiFiMain.cpp:199]: This patch seems to be working for me:
```
diff --git a/minifi_main/MiNiFiMain.cpp b/minifi_main/MiNiFiMain.cpp
index 61d49a7bc..733b1194d 100644
--- a/minifi_main/MiNiFiMain.cpp
+++ b/minifi_main/MiNiFiMain.cpp
@@ -165,12 +165,14 @@ void overridePropertiesFromCommandLine(const argparse::ArgumentParser& parser, c
   if (!parser.is_used("--schema")) {
     return std::nullopt;
   }
-  const auto& schema_path = parser.get("--schema");
-  if (schema_path == "-") {
+  const auto& schema_path_args = parser.get<std::vector<std::string>>("--schema");
+  if (schema_path_args.empty()) {
     writeJsonSchema(configure, std::cout);
     return 0;
   }
 
+  const auto& schema_path = schema_path_args[0];
+
   std::cout << "Writing json schema to " << schema_path << std::endl;
   {
     std::ofstream schema_file{schema_path};
@@ -196,8 +198,8 @@ int main(int argc, char **argv) {
     .help("Generate documentation in the specified directory");
   argument_parser.add_argument("-s", "--schema")
     .metavar("PATH")
-    .default_value("-")
-    .help("Generate JSON schema to the specified path");
+    .nargs(0, 1)
+    .help("Generate JSON schema to the specified path. If no path is specified, the schema will be printed to stdout");
 
   try {
     argument_parser.parse_args(argc, argv);
```

Comment (szaszm) [reply minifi_main/MiNiFiMain.cpp:199]: I want to keep the `-` meaning `stdout`, like most command line tools. Would you still prefer a separate empty check in that case too?

Comment (lordgamez) [reply minifi_main/MiNiFiMain.cpp:199]: We can keep both approaches, but if you would only stick with the `-` approach I'm okay with that too. It was just more intuitive to me not give any parameters if `stdout` is used. You can decide I'm okay with either option.

**Review by lordgamez — APPROVED** [review]

File: minifi_main/MiNiFiMain.cpp (hunk 1 above)
Comment (szaszm) [reply minifi_main/MiNiFiMain.cpp:199]: with the current version, since the default value is `-`, `stdout` is used when no parameters are given, as well as when `-` is given as parameter.
So both of these print to the stdout:
- `./minifi --schema`
- `./minifi --schema -`

**Comment by szaszm** [comment]
Response to Copilot low confidence comments:
1. I ensured that the callers only exit when the optional exit code is set (not nullopt).
2. I didn't read the whole codebase looking for edge cases, but I believe that this _probably_ won't cause an issue, also relying on our automated test coverage.
